docs/4.x/archiving-behavior-specification.md

```diff
@@ -0,0 +1,385 @@
+number of concurrent processes to run at a time is specified by the `--concurrent-requests-per-website` CLI option.
+
+Archive jobs for different sites are never run concurrently from within the same `core:archive` process. Sites can be
+archived concurrently but this is only possible by running multiple `core:archive` commands.
+
+**Order of Execution**
+
+Some archive invalidations have to be handled before others. Specifically:
+
+- the "All Visits" segment is run before individual segments for a period.
+- periods are run in ascending order: day periods are run before the weeks that contain them
+- invalidations are processed in descending order of start date. So the day starting on 07-26-2021 will run before
+  the week on 07-26-2021, and both will run after the day starting on 07-27-2021.
+
+**Multiple core:archive instances**
+
+As mentioned above, multiple core:archive commands can be launched in order to process sites in parallel. These runs
+do not have to be on the same server.
+
+When a core:archive instance wants to run archiving for a site, the SharedSiteIds class is used. This class manages
+a list of site IDs in an `option`. The process pulls a site ID and removes it from the list, then starts processing it.
+
+Another core:archive process may then take the next site and process that one in parallel.
+
+If `--force-idsites` is used, core:archive will not use SharedSiteIds but FixedSiteIds. This class has no knowledge of
+SharedSiteIds, so it is possible by using this we can launch archiving for the same site multiple times in parallel,
+which is generally something we want to avoid.
+
+### General Expected Behavior
+
+**invalidating today & yesterday archives**
+
+`core:archive` will automatically invalidate the archives for `today` and `yesterday` if certain criteria are met.
+
+For `today`, the archive will be automatically invalidated if there are visits for today and if the existing archive is
+older than the configured TTL (see [the previous section on archiving settings](#settings)).
+
+For `yesterday`, the archive will be automatically invalidated if the current date is a different day than the `ts_archived`
+date for the latest archive for `yesterday`. If this is true, it means the day has changed since the archive was processed,
+and there may be more visits to process. For example, if the last known archive for 2021-05-20 is calculated at 2021-05-20 20:00:00,
+but it is now 2021-05-21 00:30:00, there may be visits between 20:00:00 and 00:00:00 (midnight) to process.
+
+**invalidating custom ranges**
+
+If custom ranges are specified in the `[General] archiving_custom_ranges` INI config setting and/or
+the default period to load in Matomo is configured to be a range, they will be invalidated and
+rearchived during `core:archive`.
+
+**queued invalidations**
+
+Archives that are invalidated from system functions, for example, invalidating past data when a segment or entity changes, are
```

Comment:
here we could also mention when tracking a request for an older date? Then we remember that archive to be invalidated.

Reply:
:+1: